impl/etcd/queue/etcd.go

```diff
@@ -145,11 +170,26 @@ func (q *Queue) dequeue(key, value []byte, rev int64, cbs callbacks) error {
 		return err
 	}
 
+	// Read mutation sequence number.
+	bSequence, err := q.srvConfig.Read(q.ctx, txn, sequenceConfigKey)
+	if err != nil {
+		return err
+	}
+	sequence := binary.LittleEndian.Uint64(bSequence)
```

Comment:
It loos like these binary conversions are creeping into the code. It would be better to have a slightly custom interface with the types that we want and to do all the serialization there.